src/defrag.c

```diff
@@ -656,10 +652,7 @@ int scanLaterStreamListpacks(robj *ob, unsigned long *cursor, monotime endtime)
     static unsigned char next[sizeof(streamID)];
     raxIterator ri;
     long iterations = 0;
-    if (ob->type != OBJ_STREAM || ob->encoding != OBJ_ENCODING_STREAM) {
-        *cursor = 0;
```

Comment:
so, in fact, this line that existed for stream (that can't change encoding), was missing for the other types. :smile: